clr.alljits subset fails cross-compiling for linux-x86 on linux-x64

### Description

Cross-compiling the clr.alljits subset for linux-x86 on linux-x64 fails.

The issue seems to be related to building the win-x86 jit as commenting out this line lets the build succeed:
https://github.com/dotnet/runtime/blob/73665cc31f47264b70b4c9755ee2be1712b4f532/src/coreclr/jit/CMakeLists.txt#L595

I think the solution might be to change `TARGET_UNIX` to `HOST_UNIX` here:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/inc/clrnt.h#L836


### Reproduction Steps

Checkout repo on ubuntu 20.04, build with

```
./eng/common/cross/build-rootfs.sh x86
./eng/build.sh -ci -c Release -cross -os Linux -arch x86 -subset clr.alljits
```


### Expected behavior

Build succeeds

### Actual behavior

Build fails with the error

```
Consolidate compiler generated dependencies of target gcinfo_win_x86
  [  4%] Built target jitinterface_x86
  [  4%] Built target gcinfo_universal_arm
  Consolidate compiler generated dependencies of target eventprovider
  [  4%] Building CXX object gcinfo/CMakeFiles/gcinfo_win_x86.dir/__/gcdump/gcdump.cpp.o
  [  4%] Building CXX object gcinfo/CMakeFiles/gcinfo_win_x86.dir/__/gcdump/i386/gcdumpx86.cpp.o
  Scanning dependencies of target libunwind
  Consolidate compiler generated dependencies of target libunwind
  [  4%] Built target eventprovider
  [  4%] Building C object /runtime/artifacts/obj/external/libunwind/CMakeFiles/libunwind.dir/runtime/src/native/external/libunwind/src/os-linux.c.o
  [  4%] Built target eventing_headers
  [  4%] Building C object /runtime/artifacts/obj/external/libunwind/CMakeFiles/libunwind.dir/runtime/src/native/external/libunwind/src/mi/init.c.o
  [  4%] Building C object /runtime/artifacts/obj/external/libunwind/CMakeFiles/libunwind.dir/runtime/src/native/external/libunwind/src/mi/flush_cache.c.o
  [  4%] Building C object /runtime/artifacts/obj/external/libunwind/CMakeFiles/libunwind.dir/runtime/src/native/external/libunwind/src/mi/mempool.c.o
  [  4%] Building C object /runtime/artifacts/obj/external/libunwind/CMakeFiles/libunwind.dir/runtime/src/native/external/libunwind/src/mi/strerror.c.o
  In file included from /runtime/src/coreclr/gcdump/i386/gcdumpx86.cpp:12:
  In file included from /runtime/src/coreclr/inc/utilcode.h:44:
  /runtime/src/coreclr/inc/clrnt.h:841:16: error: redefinition of '_RUNTIME_FUNCTION'
  typedef struct _RUNTIME_FUNCTION {
                 ^
  /runtime/src/coreclr/pal/inc/pal.h:3085:16: note: previous definition is here
  typedef struct _RUNTIME_FUNCTION {
                 ^
  In file included from /runtime/src/coreclr/gcdump/i386/gcdumpx86.cpp:12:
  In file included from /runtime/src/coreclr/inc/utilcode.h:44:
  /runtime/src/coreclr/inc/clrnt.h:846:16: error: redefinition of '_DISPATCHER_CONTEXT'
  typedef struct _DISPATCHER_CONTEXT {
                 ^
  /runtime/src/coreclr/pal/inc/rt/palrt.h:1205:16: note: previous definition is here
  typedef struct _DISPATCHER_CONTEXT {
                 ^
  In file included from /runtime/src/coreclr/gcdump/gcdump.cpp:14:
  In file included from /runtime/src/coreclr/inc/utilcode.h:44:
  /runtime/src/coreclr/inc/clrnt.h:841:16: error: redefinition of '_RUNTIME_FUNCTION'
  typedef struct _RUNTIME_FUNCTION {
                 ^
  /runtime/src/coreclr/pal/inc/pal.h:3085:16: note: previous definition is here
  typedef struct _RUNTIME_FUNCTION {
                 ^
  In file included from /runtime/src/coreclr/gcdump/gcdump.cpp:14:
  In file included from /runtime/src/coreclr/inc/utilcode.h:44:
  /runtime/src/coreclr/inc/clrnt.h:846:16: error: redefinition of '_DISPATCHER_CONTEXT'
  typedef struct _DISPATCHER_CONTEXT {
                 ^
  /runtime/src/coreclr/pal/inc/rt/palrt.h:1205:16: note: previous definition is here
  typedef struct _DISPATCHER_CONTEXT {
                 ^

```

### Regression?

This built for .NET 5.0.3 (though the resulting product didn't function).  

I think the regression may have been in https://github.com/dotnet/runtime/pull/41126 but I'm not sure about this.

The build fails in .NET 6.0.3 also.

### Known Workarounds

I think the solution might be to change `TARGET_UNIX` to `HOST_UNIX` here:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/inc/clrnt.h#L836

### Configuration

Attempting to cross-compile `main` commit `89b034d8e3`.

Using Ubuntu 20.04 x64 VM under Windows 10 x64 hyper-v.

### Other information

_No response_

--- a/src/coreclr/inc/clrnt.h
+++ b/src/coreclr/inc/clrnt.h
@@ -833,7 +833,7 @@ RtlVirtualUnwind_Unsafe(
 //
 
 #ifdef TARGET_X86
-#ifndef TARGET_UNIX
+#ifndef HOST_UNIX
 //
 // x86 ABI does not define RUNTIME_FUNCTION. Define our own to allow unification between x86 and other platforms.
 //
@@ -847,7 +847,7 @@ typedef struct _DISPATCHER_CONTEXT {
     _EXCEPTION_REGISTRATION_RECORD* RegistrationPointer;
 } DISPATCHER_CONTEXT, *PDISPATCHER_CONTEXT;
 #endif // HOST_X86
-#endif // !TARGET_UNIX
+#endif // !HOST_UNIX
 
 #define RUNTIME_FUNCTION__BeginAddress(prf)             (prf)->BeginAddress
 #define RUNTIME_FUNCTION__SetBeginAddress(prf,addr)     ((prf)->BeginAddress = (addr))

